src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -102,64 +113,67 @@ public NuGetProject DefaultNuGetProject
                 throw new ArgumentNullException(nameof(projectSystemFactory));
             }
 
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _serviceProvider = serviceProvider;
             _projectSystemCache = projectSystemCache;
             _projectSystemFactory = projectSystemFactory;
+            _logger = logger;
+        }
 
-            _dte = ServiceLocator.GetInstance<DTE>();
-            _vsSolution = ServiceLocator.GetGlobalService<SVsSolution, IVsSolution>();
-            _vsMonitorSelection = ServiceLocator.GetGlobalService<SVsShellMonitorSelection, IVsMonitorSelection>();
+        public async Task InitializeAsync(IAsyncServiceProvider site)
+        {
```

Comment:
`Due to high risk of a deadlock in VS any threading work in MEF is forbidden.` 
how about adding jtf.RunAsync inside MEF which shouldn't risk of a deadlock in VS, is that also forbidden?
Delay attaching events is fine, but I'm just afraid if there would be a case where they never attached. Just think through that otherwise it's good.